control_panel/Control_Panel_Waypoints_View.py

```diff
@@ -257,62 +251,15 @@ def _send(self):
                                        "There are no vehicles with waypoints.")
             return
 
-        # Create a progress dialog and send the waypoints to the vehicles.
-        self._progress = QtGui.QProgressDialog(self._controller.central_widget)
-        self._progress.setMinimum(0)
-        self._progress.setMaximum(total)
-        self._progress.setWindowModality(QtCore.Qt.WindowModal)
-        self._progress.setMinimumDuration(0)
-        self._progress.setCancelButtonText("Cancel")
-        self._progress.canceled.connect(lambda: self._cancel())
-        self._progress.setWindowTitle("Sending waypoints")
-        self._progress.setLabelText("Initializing...")
-        self._progress.open()
-
-        self._labels = dict([(vehicle, "") for vehicle in waypoints])
-
-        self._retry_counts = dict([(vehicle, self._max_retries) for vehicle in waypoints])
-        self._indexes = dict([(vehicle, -1) for vehicle in waypoints])
-
-        self._timers = {}
-        for vehicle in waypoints:
-            timer = QtCore.QTimer()
-            timer.setInterval(self._retry_interval * 1000)
-            timer.setSingleShot(True)
-            # Bind timeout signal to retry for the current vehicle.
-            timer.timeout.connect(partial(self._retry, vehicle))
-            self._timers[vehicle] = timer
-
-        self._waypoints = waypoints
-        self._total = total
-        for vehicle in self._waypoints:
-            self._send_clear(vehicle)
-
-    def _send_clear(self, vehicle):
-        packet = XBee_Packet()
-        packet.set("specification", "waypoint_clear")
-        packet.set("to_id", vehicle)
-
-        self._controller.xbee.enqueue(packet, to=vehicle)
-
-        self._set_label(vehicle, "Clearing old waypoints")
-        self._timers[vehicle].start()
-
-    def _send_one(self, vehicle):
-        index = self._indexes[vehicle]
-        if len(self._waypoints[vehicle]) <= index:
-            # Enqueue a packet indicating that waypoint sending
-            # for this vehicle is done.
-            packet = XBee_Packet()
-            packet.set("specification", "waypoint_done")
-            packet.set("to_id", vehicle)
-            self._controller.xbee.enqueue(packet, to=vehicle)
-
-            self._update_value()
-            return
-
-        waypoint = self._waypoints[vehicle][index]
+        sender = Control_Panel_XBee_Sender(self._controller, "waypoint",
```

Comment:
Can we pass an object here instead of all these arguments? It helps keeping this readable.